embed-builder/src/builder.rs

```diff
@@ -15,121 +16,139 @@ use twilight_model::channel::embed::{
-///
-/// This is returned from [`EmbedBuilder::title`].
-#[derive(Clone, Debug, Eq, PartialEq)]
-pub enum EmbedTitleError {
+    /// Name is empty.
+    FieldNameEmpty {
+        /// Provided name. Although empty, the same owned allocation is
+        /// included.
+        name: String,
+        /// Provided value.
+        value: String,
+    },
+    /// Name is longer than 256 UTF-16 code points.
+    FieldNameTooLong {
+        /// Provided name.
+        name: String,
+        /// Provided value.
+        value: String,
+    },
+    /// Value is empty.
+    FieldValueEmpty {
+        /// Provided name.
+        name: String,
+        /// Provided value. Although empty, the same owned allocation is
+        /// included.
+        value: String,
+    },
+    /// Value is longer than 1024 UTF-16 code points.
+    FieldValueTooLong {
+        /// Provided name.
+        name: String,
+        /// Provided value.
+        value: String,
+    },
+    /// Footer text is empty.
+    FooterTextEmpty {
+        /// Provided text. Although empty, the same owned allocation is
+        /// included.
+        text: String,
+    },
+    /// Footer text is longer than 2048 UTF-16 code points.
+    FooterTextTooLong {
+        /// Provided text.
+        text: String,
+    },
     /// Title is empty.
-    Empty {
+    TitleEmpty {
         /// Provided title. Although empty, the same owned allocation is
         /// included.
         title: String,
     },
     /// Title is longer than 256 UTF-16 code points.
-    TooLong {
+    TitleTooLong {
         /// Provided title.
         title: String,
     },
+    /// The total content of the embed is too large.
+    ///
+    /// Refer to [`EmbedBuilder::EMBED_LENGTH_LIMIT`] for more information about
+    /// what goes into this limit.
+    ///
+    /// [`EmbedBuilder::EMBED_LENGTH_LIMIT`]: struct.EmbedBuilder.html#associatedconstant.EMBED_LENGTH_LIMIT
+    TotalContentTooLarge {
+        /// The total length of the embed.
+        length: usize,
+    },
+    /// Too many fields were provided.
+    ///
+    /// Refer to [`EmbedBuilder::EMBED_FIELD_LIMIT`] for more information about
+    /// what the limit is.
+    ///
+    /// [`EmbedBuilder::EMBED_FIELD_LIMIT`]: struct.EmbedBuilder.html#associatedconstant.EMBED_FIELD_LIMIT
+    TooManyFields {
+        /// The provided fields.
+        fields: Vec<EmbedField>,
+    },
 }
 
-impl Display for EmbedTitleError {
+impl Display for EmbedError {
     fn fmt(&self, f: &mut Formatter<'_>) -> FmtResult {
         match self {
-            Self::Empty { .. } => f.write_str("the title is empty"),
-            Self::TooLong { .. } => f.write_str("the title is too long"),
+            Self::AuthorNameEmpty { .. } => f.write_str("the author name is empty"),
+            Self::AuthorNameTooLong { .. } => f.write_str("the author name is too long"),
+            Self::ColorNotRgb { color } => {
+                f.write_fmt(format_args!("the color {} is invalid", color))
```

Comment:
Doesn’t this do the same thing? Both go through the fmt machinery, I think.
```suggestion
                write!(f, "the color {} is invalid", color)
```

Reply:
Twilight tends to not use macros in order to support editors better

Reply:
Isn’t `format_args!` a macro too though? I don’t see how it and a function are better then just `write!()`

Reply:
Resolving because this doesn’t matter :D